server/coderadar-webapp/src/main/java/org/wickedsource/coderadar/security/service/PasswordService.java

```diff
@@ -0,0 +1,36 @@
+package org.wickedsource.coderadar.security.service;
+
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service for working with passwords, for example hashing or verification.
+ */
+@Service
+public class PasswordService {
+
+    private final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+
+
+    /**
+     * hash the password with "bcrypt"
+     *
+     * @param password the password to be hashed
+     * @return hashed password as hexadecimal
+     */
+    public String hash(String password) {
+        return passwordEncoder.encode(password);
+    }
+
+    /**
+     * checks, if the passwords match
+     *
+     * @param rowPassword     the password to be verified
+     * @param encodedPassword encoded the password as hexadecimal
+     * @return <code>true</code>, if the password match, false otherwise
+     */
+    boolean verify(String rowPassword, String encodedPassword) {
```

Comment:
_rawPassword_ would be the right name according to  [PasswordEncoder](http://docs.spring.io/spring-security/site/docs/current/apidocs/org/springframework/security/crypto/password/PasswordEncoder.html), that is used in the service.